cpp/dolfinx/common/MPI.cpp

```diff
@@ -79,14 +79,32 @@ int dolfinx::MPI::rank(const MPI_Comm comm)
   return rank;
 }
 //-----------------------------------------------------------------------------
-//-----------------------------------------------------------------------------
 int dolfinx::MPI::size(const MPI_Comm comm)
 {
   int size;
-  MPI_Comm_size(comm, &size);
+  int ierr = MPI_Comm_size(comm, &size);
+  dolfinx::MPI::assert_and_throw(comm, ierr);
   return size;
 }
 //-----------------------------------------------------------------------------
+void dolfinx::MPI::assert_and_throw(MPI_Comm comm, int error_code)
```

Comment:
I don't think the function name is quite right. `assert` might imply that the check is skipped in non-debug mode, and the function doesn't 'throw' an exception; it just aborts.